tests/integration/security/reachability/mtls_test.go

```diff
@@ -0,0 +1,192 @@
+	bApp, _ := appsInstance.GetAppOrFail("b", t).(apps.KubeApp)
+
+	headlessApp, _ := appsInstance.GetAppOrFail("headless", t).(apps.KubeApp)
+	// App without sidecar.
+	nakedApp, _ := appsInstance.GetAppOrFail("t", t).(apps.KubeApp)
+
+	testCases := []struct {
+		configFile  string
+		namespace   string
+		connections []connect.Connection
+	}{
+		{
+			configFile: "global-mtls-on.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            aApp,
+					To:              headlessApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-permissive.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-off.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "single-port-mtls-on.yaml",
+			namespace:  appsInstance.Namespace().Name(),
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+	}
+
+	for _, c := range testCases {
+		namespace := c.namespace
+		if len(namespace) == 0 {
+			namespace = istioCfg.SystemNamespace
+		}
+
+		policy := policy.ApplyPolicyFile(t, env, namespace, c.configFile)
```

Comment:
Can you try to change this to 

defer policy.ApplyPolicyFile(...).TearDown()

It was the reason why I made apply function return the struct, but somehow I forgot to apply in the previous code.

Reply:
Changed to: 		policy := policy.ApplyPolicyFile(t, env, namespace, c.configFile)
		defer policy.TearDown()